packages/material-ui/src/RadioGroup/RadioGroup.js

```diff
@@ -2,92 +2,40 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import warning from 'warning';
 import FormGroup from '../FormGroup';
-import { createChainedFunction, find } from '../utils/helpers';
-
-class RadioGroup extends React.Component {
-  radios = [];
-
-  constructor(props) {
-    super();
-    this.isControlled = props.value != null;
-
-    if (!this.isControlled) {
-      this.state = {
-        value: props.defaultValue,
-      };
-    }
-  }
-
-  focus = () => {
-    if (!this.radios || !this.radios.length) {
-      return;
+import SelectableGroup from '../internal/SelectableGroup';
+
+function RadioGroup(props) {
+  const { children, defaultValue, name, onChange, value: valueProp, ...other } = props;
+  const { current: isUncontrolled } = React.useRef(valueProp == null);
+  const [value, setValue] = React.useState(() => {
+    if (isUncontrolled) {
+      return defaultValue;
     }
+    return null;
+  });
 
-    const focusRadios = this.radios.filter(n => !n.disabled);
-
-    if (!focusRadios.length) {
-      return;
-    }
-
-    const selectedRadio = find(focusRadios, n => n.checked);
-
-    if (selectedRadio) {
-      selectedRadio.focus();
-      return;
-    }
-
-    focusRadios[0].focus();
-  };
+  if (valueProp !== undefined && valueProp !== value) {
+    setValue(valueProp);
+  }
 
-  handleChange = event => {
-    if (!this.isControlled) {
-      this.setState({
-        value: event.target.value,
-      });
+  const handleChange = (event, selected) => {
+    if (isUncontrolled) {
+      setValue(selected);
     }
 
-    if (this.props.onChange) {
-      this.props.onChange(event, event.target.value);
+    if (onChange) {
+      onChange(event, selected);
     }
   };
 
-  render() {
-    const { children, name, value: valueProp, onChange, ...other } = this.props;
-
-    const value = this.isControlled ? valueProp : this.state.value;
-    this.radios = [];
-
-    return (
+  return (
+    <SelectableGroup onChange={handleChange} value={value} name={name} exclusive>
```

Comment:
As we are using the `exclusive` mode. Can't we save SelectableGroup bundle size by using a context forwarding direct? I mean, I would anticipate a straightforward logic without SelectableGroup, but by using the context. Do we need this abstraction? Maybe it's only suited for the ToggleButton? I was thinking of doing something similar with the Tabs component. I'm not I guess React context is all we need?